apps/frontend/src/components/ItemModal.tsx

```diff
@@ -0,0 +1,118 @@
+import React from 'react';
+import {
+  HStack,
+  VStack,
+  Text,
+  Card,
+  CardBody,
+  Stack,
+  AspectRatio,
+  Image,
+  Divider,
+  Tag,
+  Wrap,
+  WrapItem,
+  Button,
+  Link,
+} from '@chakra-ui/react';
+
+import GeneralModal from './GeneralModal';
+import { ExternalLinkIcon } from '@chakra-ui/icons';
+
+
+type ItemModalHeaderProps = {
+    name: string;
+}
+
+function ItemModalHeader({ name }: ItemModalHeaderProps) {
+  return (
+    <HStack w={'100%'} justify={'center'}>
+      <Text fontSize={'xl'}>{name}</Text>
+    </HStack>
+  );
+}
+
+
+type ItemModalBodyProps = {
+    id: number;
+    name: string;
+    description: string;
+    imageUrl: string;
+    url: string
+    tags: string[];
+}
+
+function ItemModalBody({ id, name, description, imageUrl, url, tags }: ItemModalBodyProps) {
+  return (
+    <VStack>
+      <Card>
+        <CardBody>
+          <Stack spacing={4}>
+
+            <AspectRatio ratio={ 16 / 9 }>
+              <Image src={imageUrl || 'https://via.placeholder.com/150'} objectFit={'cover'} borderRadius={5}/>
+            </AspectRatio>
+
+            <Text>{description}</Text>
+
+            <Divider />
+
+            <Link href={url} isExternal> {name} <ExternalLinkIcon mx="2px" /> </Link>
+
+            <Divider />
+
+            <Wrap spacing={0}>
+              {tags.map(tag => (
+                <WrapItem key={id}>
+                  <Tag size="lg" m={1} backgroundColor={'brand.100'} borderRadius="15px">
+                    {tag}
+                  </Tag>
+                </WrapItem>
+              ))}
+            </Wrap>
+
+          </Stack>
+        </CardBody>
+      </Card>
+    </VStack>
+  );
+}
+
+
+type ItemModalFooterProps = {
+    onAddItem: () => void;
+    onRemoveItem: () => void;
+}
+
+function ItemModalFooter({ onAddItem, onRemoveItem } : ItemModalFooterProps) {
+  return (
+    <HStack w={'100%'} spacing={4}>
+      <Button flex={1} variant={'solid'} colorScheme="brand" onClick={onAddItem}>Add Item</Button>
+      <Button flex={1} variant={'outline'} colorScheme="brand" onClick={onRemoveItem}>Remove Item</Button>
```

Comment:
This should be a single button that says either "Add to my Wishlist" or "Remove from my Wishlist". Except when user is viewing their own collection, then it could only say "Remove from my Collection"

Reply:
Done. Update the Footer Props so it can take it a variable number of buttons.

<img width="597" alt="Screenshot 2024-03-29 at 16 07 26" src="https://github.com/Geek-Collectors-Network/geek-collectors-network/assets/97256979/e9ecd174-5c06-4c5c-9479-b381cb229dad">

<img width="586" alt="Screenshot 2024-03-29 at 16 08 43" src="https://github.com/Geek-Collectors-Network/geek-collectors-network/assets/97256979/cd243f8b-1734-4c2e-a6f1-c59a503e20ae">